src/mono/mono/component/mini-wasm-debugger.c

```diff
@@ -198,7 +199,7 @@ assembly_loaded (MonoProfiler *prof, MonoAssembly *assembly)
 		return;
 	}
 
-	if (mono_wasm_assembly_already_added(assembly->aname.name))
+	if (mono_bundled_resources_get_assembly_resource_values(assembly->aname.name, NULL, NULL, NULL, NULL))
```

Comment:
I think there is a small change in behavior here, `mono_wasm_assembly_already_added `checks if there is any assemblies added `assembly_count != 0` and compare without file extension against existing registered assemblies (since assembly->aname.name might not include file extension).

That means, change will not find registered assemblies in case they are registered with .dll file extension and assembly->aname.name doesn't include .dll file extension, while current implementation would handle that case.

Another case to consider is when we only have pdb parts registered for the assembly, but looking at mono_bundled_resources_get_assembly_resource_values it should return false in case where assembly data/size is NULL/0, so looks like that case is handled.

Since the pdb gets registered on the

Reply:
Good point, it seems like `assembly->aname.name` should never have a known extension (`.dll`, `.wasm`, `.webcil`) in order for it to match extensionless registered assembly names.

Not sure why exactly we can say that `assembly->aname.name` will not have the `.dll` (or other) extension, the assemblies that are looped through and passed as `assembly` in this function are added through `mono_alc_add_assembly`, which it seems from the `sre.c` path are added via `ves_icall_AssemblyBuilder_basic_init` (`basic_init`) so somehow that shouldn't have the `.dll`?
The other instance of `mono_alc_add_assembly` is via `mono_domain_fire_assembly_load` but that path is also not clear to me how the extensions get dropped.

That seems fragile in the case where a registered assembly contains a `.` but not a known extension, i.e. if the registered name for a resource like `WebAssembly.Browser.StartupHook.Test.dll` was the filename without extension `WebAssembly.Browser.StartupHook`. In that case, the `assembly->aname.name` would probably be `WebAssembly.Browser.StartupHook`, but the comparison would have been against `WebAssembly.Browser` and there would never have been a match because of the length comparison `entry_name_minus_extn_len == strlen(assembly_name)`.

In anycase, I think then if its granted that `assembly->aname.name` will always be the name without extension (the original assumption), we can first check if `assembly->aname.name` is in the bundle and then check if `assembly->aname.name` + `.dll` is in the bundle. That would be fragile if both `<assemblyname>` and `<assemblyname>.dll` are both registered in the bundle, but `mono_wasm_assembly_already_added` is already prone to that scenario.

Reply:
The assembly->aname->name is loaded from assembly image in `mono_assembly_fill_assembly_name_full` and that will just include the name of the assembly without file extensions, so if that field is used to try to locate a bundled assembly registered with .dll, we would need to append it as part of searching.